Only register floatingIP for external loadbalancer service

FengyunPan opened the pull request:
If the user has provided the floating-ip options, then it's safe
to assume they want (only) the floating-ip to be the ingress IP;
if they have not provided floating-ip options, then the LB IP is
the only relevant value.


**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
Fix #52566

**Release note**:
```release-note
Only register floatingIP into Loadbalancer ingress field for external loadbalancer service
```

dims commented:
/retest

dims commented:
/test all

anguslees commented:
An idea:  Instead of adding yet another annotation, could we just automatically intuit this from whether there is a floating-ip?

In other words, I think the _goal_ is to publish the intended ingress IP that consumers should use to access the LB service.  _If_ the user has provided the floating-ip options, then I think it's safe to assume they want (only) the floating-ip to be the ingress IP; if they have _not_ provided floating-ip options, then the LB IP is the only relevant value.

Commit:
Only register floatingIP for external loadbalancer service

If the user has provided the floating-ip options, then it's safe
to assume they want (only) the floating-ip to be the ingress IP;
if they have not provided floating-ip options, then the LB IP is
the only relevant value.
Fix #52566

FengyunPan commented:
@anguslees Good idea, have updated it.

anguslees commented on pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go:
```diff
@@ -1472,7 +1470,9 @@ func (lb *LbaasV1) EnsureLoadBalancer(clusterName string, apiService *v1.Service
 		}
 
-		status.Ingress = append(status.Ingress, v1.LoadBalancerIngress{IP: floatIP.FloatingIP})
+		status.Ingress = []v1.LoadBalancerIngress{{IP: floatIP.FloatingIP}}
```
Aside: I see this code is structured ever-so-slightly differently to the LbaasV2 version (with different corner cases around whether the floating IP already exists), which is unfortunate.  It would be nice if there was less duplication/variation here - otoh the LbaasV1 version could probably also be declared obsolete and just removed...

FengyunPan replied:
Yeah, there are many different places Between LbaasV1 and LbaasV2, LbaasV1 has been  deprecated on OpenStack, so it is better to remove LbaasV1.

anguslees commented:
This is a user-visible change, so should have a release-note (imo).

FengyunPan commented:
/retest

dims commented:
/test all

dims commented:
/approve
/lgtm
